system/graph-node/wrapper/src/lib.rs

```diff
@@ -0,0 +1,63 @@
+pub mod wrap;
+
+pub use wrap::*;
+use crate::imported::{ HttpModule, ArgsPost };
+use polywrap_wasm_rs::JSON;
+use serde::Deserialize;
+
+#[derive(Deserialize)]
+struct ErrorLocation {
+    column: i32,
+    line: i32
+}
+
+#[derive(Deserialize)]
+struct Error {
+    locations: Option<Vec<ErrorLocation>>,
+    message: String,
+}
+
+#[derive(Deserialize)]
+struct RequestError {
+    errors: Vec<Error>,
+}
+
+pub fn query_subgraph(args: ArgsQuerySubgraph, env: Env) -> String {
+    let ArgsQuerySubgraph { subgraph_author, subgraph_name, query } = args;
+
+    // prepare http request
+    let url: String = format!("{}/subgraphs/name/{}/{}", &env.provider, &subgraph_author, &subgraph_name);
```

Comment:
I'm wondering what's the reason for doing it like this (i.e. with author and name passed in, the base URL as an environment setting)? Why not just have the Subgraph URL passed in? 

The reason I'm asking this is because Subgraphs can be self-hosted; there are more hosting providers than The Graph's nowadays; The Graph itself is split into Hosted Service and Subgraph Studio.

Reply:
That's good input. Could you give some examples of subgraph URIs with different structures? Or a link to a resource where I can learn more?

Reply:
Thanks for getting back so quick!

Disclaimer: I'm not a Subgraph/GraphQL expert or have any association.

This an explorer for the new Subgraph Studio: https://thegraph.com/explorer

A URL example: `https://gateway.thegraph.com/api/[api-key]/subgraphs/id/FDD65maya4xVfPnCjSgDRBz6UBWKAcmGtgY6BmUueJCg` Note the `[api-key]`.

Like I previously mentioned, the Subgraphs can also be self-hosted or hosted through other providers. There's really no predefined rule how to define the API endpoint.

You might get some inspiration out of [The Graph Client Tools](https://github.com/graphprotocol/graph-client). 

I'd also like to point out that this wrapper doesn't necessarily have to be Subgraph specific -- it could be a generic GraphQL client. But I totally understand if you don't want to go down the rabbit hole of creating a generic client because it could be made to support all sorts of features and edge-cases. The Graph/[The Guild](https://the-guild.dev/) people could probably help out. I'd say what's definitely currently missing is the separation of `variables` from `query`: https://graphql.org/learn/serving-over-http/#get-request

Hope this helps a bit.

Reply:
https://github.com/polywrap/integrations/pull/133